pkg/kwokctl/runtime/binary/cluster.go

```diff
@@ -742,6 +744,61 @@ func (c *Cluster) LogsFollow(ctx context.Context, name string, out io.Writer) er
 	return nil
 }
 
+// CollectLogs returns the logs of the specified component.
+func (c *Cluster) CollectLogs(ctx context.Context, name string, dir string) error {
+	logger := log.FromContext(ctx)
+	var errs []error
+
+	path := filepath.Join(dir, consts.RuntimeTypeBinary+"-info.txt")
+	f, err := os.Create(path)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		_, err = f.WriteString(fmt.Sprintf("%s %s", rt.GOOS, rt.GOARCH))
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	defer func() {
+		err = f.Close()
+		if err != nil {
+			logger.Error("Failed to close file", err)
+			_ = os.Remove(path)
+		}
+	}()
+
+	cfg, err := c.Config(ctx)
+	if err != nil {
+		errs = append(errs, err)
+		return errors.Join(errs...)
+	}
```

Comment:
```suggestion
	cfg, err := c.Config(ctx)
	if err != nil {
		return err
	}
```

Move to top of function